### ISSUE TITLE
Dark mode updates after #1089

### ISSUE THREAD

**Issue by christianvuerings** [pinterest/gestalt#1102]
Follow ups from @ajmoo in #1089:

https://github.com/pinterest/gestalt/pull/1089#issuecomment-666822334
> I've got some notes so far…
> 
> * Secondary (lightGray) `Buttons` should be #404040, hover state #535353 and active state #666666
> * `IconButton` should be #535353 hover state and #666666 active state
> * Noticing a bug with `button`: after clicking, the hover state color doesn't clear.
> * I don't love the blue/red+dark text pairing on , but don't know how to resolve this one right now. cc @anniemochi
> * The white shadows on `DatePicker` and elsewhere look off to me. @anniemochi is it okay to keep shadows black in dark mode?
> * Should the accessibility focus indicator have transparency? Looks wrong against a dark background. How does #9CC2FB look? @anniemochi
> * hover state for `DatePicker` should be #535353 to align with button hover state
> * Are you addressing `SelectList`, `TextField`, `TextArea`, and `SearchField` in this PR?
> * `SegmentedControl` doesn't have a hover state. @anniemochi is that ok? More noticeable in dark mode.
> * `Tab` hover state should be #535353

https://github.com/pinterest/gestalt/pull/1089#issuecomment-666843999

> Oh yeah, and our Primary red `button`…
> #CF001F hover
> #B8001C active
> 
> (values are basically adding 10% black and 20% black on top of the red)

**Comment by jennyscript** [comment]
Responses to questions on the other ticket here https://github.com/pinterest/gestalt/pull/1089#issuecomment-669600202

**Comment by jennyscript** [comment]
I've got fixes for a few of these on #1129. Just a few questions for the remaining ones:

@ajmoo You mentioned a bug with button. I think this is the `:focus` state of the button from. Is this what you're seeing and if so should it be removed?
<img width="1377" alt="Screen Shot 2020-08-06 at 5 00 54 PM" src="https://user-images.githubusercontent.com/1767822/89594433-9cc85700-d806-11ea-89fc-bdbee7953be4.png">

@anniemochi I've updated the blue Buttons to have white text and the red ones already did. What are the other places to update?
